Ловим события

DimonDem123 opened the pull request:
Хорошего дня

DimonDem123 committed:
BJS2-79242
Сделал проверки на пустные значения

DimonDem123 committed:
BJS2-79721
Потыкал проверил все работает коректно.
Хорошего дня

DimonDem123 committed:
Update SpellEvent.java

DimonDem123 committed:
Update HogwartsSpells.java

DimonDem123 committed:
BJS2-79721
Потыкал проверил все работает коректно.
Хорошего дня

DimonDem123 committed:
BJS2-79721
Потыкал проверил все работает коректно. Все мои тесты прошла
Хорошего дня

zaqqaz226 commented on src/main/java/school/faang/bjs279721/SpellEvent.java:
```diff
@@ -0,0 +1,32 @@
+    private String action;
+
+    @Override
+    public boolean equals(Object o) {
```
А зачем тут переопределение? Ты же не используешь его как ключ. А вот если  был ключом, тогда имело бы смысл и то проще было бы накинуть @Data или использовать не класс, а record )

zaqqaz226 replied:
Решил оставить? Почему?

DimonDem123 replied:
Сори не заметил этот коментарий вот и оставил

zaqqaz226 commented on src/main/java/school/faang/bjs279721/Main.java:
```diff
@@ -0,0 +1,30 @@
+                System.out.println(action);
+            }
+        } else {
+            System.out.println("JJ");
```
Странный вывод в консоль

DimonDem123 replied:
Я исключения хоть и изучал, но на практике еще не пользовался, пока простительно, еще не особо разбираюсь как правильно писать учитывая все ошибки

zaqqaz226 replied:
Понимаю, но давай сразу делать правильно. Если хочешь тут выбросить исключение, то можно написать throw new SomeException("Message"); Только выбери исключение, которое тут подойдет по смыслу.

zaqqaz226 commented on src/main/java/school/faang/bjs279721/HogwartsSpells.java:
```diff
@@ -0,0 +1,88 @@
+        if (eventType == null && actionDescription == null) {
+            if (eventType.isEmpty() && actionDescription.isEmpty()) {
+                log.warn("Пустое значение");
+                return;
```
Можно и ошибку выкинуть. У нас же не должно было прийти пустое значение, значит это не характерное поведение приложения, значит можно и выкинуть исключение.

DimonDem123 replied:
Знаю что можно. Но я пока не научился ими нормлаьно пользоваться

zaqqaz226 replied:
Давай пробовать, если что, я подскажу что исправить. Напиши тут выброс исключения.

zaqqaz226 commented on src/main/java/school/faang/bjs279721/HogwartsSpells.java:
```diff
@@ -0,0 +1,88 @@
+    public String getSpellEventById(int id) {
+        if (id < 0) {
+            log.warn("Отрицательное значение не проходит");
+            return null;
```
Тут тоже можно и исключение бросить.

zaqqaz226 replied:
И тут тоже, нужно исключение. Как раз потренируешься.

zaqqaz226 commented on src/main/java/school/faang/bjs279721/HogwartsSpells.java:
```diff
@@ -0,0 +1,88 @@
+            return null;
+        }
+
+        if (spellById.containsKey(id)) {
```
Ты здесь сначала проверяешь наличие, а потом пытаешься гетнуть. Если есть то отдаешь, если нет то пернешь null. Возможно это сделать иначе )

DimonDem123 committed:
Merge remote-tracking branch 'origin/Dum3-BJS2-79721' into Dum3-BJS2-79721

DimonDem123 committed:
BJS2-79721
Добавил исключения
и чутка переделал проверку на значения

DimonDem123 committed:
BJS2-79877
Исправил вывод. Убрал коментарии добавил лог так где указали, может что-то не заметил, но смысл уловил для чего исправления

DimonDem123 committed:
Merge remote-tracking branch 'origin/Dum3-BJS2-79877' into Dum3-BJS2-79721

DimonDem123 committed:
BJS2-79877
добавл @Data вместо ручного переопределения equals and hashCode

DimonDem123 committed:
BJS2-79877
добавл @Data вместо ручного переопределения equals and hashCode
